languages/java-cpg/src/main/java/de/jplag/java_cpg/CpgAdapter.java

```diff
@@ -0,0 +1,128 @@
+        }
+    }
+
+    /**
+     * Registers the given transformations to be applied in the transformation step.
+     * @param transformations the transformations
+     */
+    public void addTransformations(GraphTransformation[] transformations) {
+        Arrays.stream(transformations).forEach(this::addTransformation);
+    }
+
+    /**
+     * Clears all non-{@link ExecutionPhase#OBLIGATORY} transformations from the pipeline.
+     */
+    public void clearTransformations() {
+        AstTransformationPass.clearTransformations();
+        CpgTransformationPass.clearTransformations();
+    }
+
+    private <T extends Pass<?>> KClass<T> getKClass(Class<T> javaPassClass) {
+        return JvmClassMappingKt.getKotlinClass(javaPassClass);
+    }
+
+    /**
+     * Sets <code>reorderingEnabled</code>. If true, statements may be reordered.
+     * @param enabled value for reorderingEnabled.
+     */
+    public void setReorderingEnabled(boolean enabled) {
+        this.reorderingEnabled = enabled;
+    }
+
+    private void setTokenList(List<Token> tokenList) {
+        this.tokenList = tokenList;
+    }
+
+    /* package-private */ TranslationResult translate(Set<File> files) throws ParsingException, InterruptedException {
+        InferenceConfiguration inferenceConfiguration = InferenceConfiguration.builder().guessCastExpressions(true).inferRecords(true)
+                .inferDfgForUnresolvedCalls(true).build();
+
+        TranslationResult translationResult;
+        TokenizationPass.Companion.setCallback(CpgAdapter.this::setTokenList);
+        try {
+            TranslationConfiguration.Builder configBuilder = new TranslationConfiguration.Builder().inferenceConfiguration(inferenceConfiguration)
+                    .sourceLocations(files.toArray(new File[] {})).registerLanguage(new JavaLanguage());
+
+            List<Class<? extends Pass<?>>> passClasses = new ArrayList<>(List.of(TypeResolver.class, TypeHierarchyResolver.class,
+                    ImportResolver.class, SymbolResolver.class, PrepareTransformationPass.class, FixAstPass.class, DynamicInvokeResolver.class,
+                    FilenameMapper.class, AstTransformationPass.class, EvaluationOrderGraphPass.class,  // creates
+                    // EOG
+                    DfgSortPass.class, CpgTransformationPass.class, TokenizationPass.class));
+
+            if (!reorderingEnabled)
+                passClasses.remove(DfgSortPass.class);
+
+            for (Class<? extends Pass<?>> passClass : passClasses) {
+                configBuilder.registerPass(getKClass(passClass));
+            }
+
+            translationResult = TranslationManager.builder().config(configBuilder.build()).build().analyze().get();
+
+        } catch (InterruptedException e) {
+            throw e;
+        } catch (ExecutionException | ConfigurationException e) {
+            throw new ParsingException(List.copyOf(files).getFirst(), e);
```

Comment:
Because any one element can not be gotten from the `Set<File>` `files`. I agree that this is not very elegant.
Maybe replace it with `files.stream().findAny().orElse(null)`?